File: app/src/main/java/com/kidozh/discuzhub/entities/ViewHistory.java
package com.kidozh.discuzhub.entities;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.Date;
import java.util.Objects;

@Entity
public class ViewHistory {
    @PrimaryKey(autoGenerate = true)
    private int id;

    public String avatarURL;
    public String name;
    public int belongedBBSId;
    public String description;
    public int type;
    public int fid;
    public int tid;
    public Date recordAt;
    @Ignore
    public final static int VIEW_TYPE_FORUM = 0, VIEW_TYPE_THREAD = 1, VIEW_TYPE_USER_PROFILE = 2;


    public ViewHistory(String avatarURL, String name, int belongedBBSId, String description, int type, int fid, int tid, Date recordAt) {

        this.avatarURL = avatarURL;
        this.name = name;
        this.belongedBBSId = belongedBBSId;
        this.description = description;
        this.type = type;
        this.fid = fid;
        this.tid = tid;
        this.recordAt = recordAt;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ViewHistory that = (ViewHistory) o;
        return id == that.id &&
                belongedBBSId == that.belongedBBSId &&
                type == that.type &&
                fid == that.fid &&
                tid == that.tid &&
                avatarURL.equals(that.avatarURL) &&
                name.equals(that.name) &&
                description.equals(that.description) &&
                recordAt.equals(that.recordAt);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, avatarURL, name, belongedBBSId, description, type, fid, tid, recordAt);
    }
}

File: app/src/main/java/com/kidozh/discuzhub/entities/ViewHistory.kt
package com.kidozh.discuzhub.entities

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.util.*

@Entity
class ViewHistory(avatarURL: String, name: String, belongedBBSId: Int, description: String, type: Int, fid: Int, tid: Int, recordAt: Date) {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    @JvmField
    var avatarURL = ""
    @JvmField
    var name = ""
    @JvmField
    var belongedBBSId: Int
    @JvmField
    var description = ""
    @JvmField
    var type: Int
    @JvmField
    var fid: Int
    @JvmField
    var tid: Int
    @JvmField
    var recordAt: Date
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ViewHistory
        return id == that.id && belongedBBSId == that.belongedBBSId && type == that.type && fid == that.fid && tid == that.tid &&
                (avatarURL == that.avatarURL) && name == that.name && description == that.description && recordAt == that.recordAt
    }

    override fun hashCode(): Int {
        return Objects.hash(id, avatarURL, name, belongedBBSId, description, type, fid, tid, recordAt)
    }

    companion object {
        @Ignore
        const val VIEW_TYPE_FORUM = 0

        @Ignore
        const val VIEW_TYPE_THREAD = 1

        @Ignore
        const val VIEW_TYPE_USER_PROFILE = 2
    }

    init {
        this.avatarURL = avatarURL
        this.name = name
        this.belongedBBSId = belongedBBSId
        this.description = description
        this.type = type
        this.fid = fid
        this.tid = tid
        this.recordAt = recordAt
    }
}